### PR TITLE
Encode log data to ascii string using c string pointer

### PR DESCRIPTION
**Pull request by memoto**
Swift 6 String's `init?(data: Data, encoding: String.Encoding)` fails validation during encoding `utf8` data to `ascii` and returns `nil`.

As proposed solution we inspect memory buffer bound to `UInt8/CChar` bytes, assuming they are null-terminted which allows us to produce a new string by simply copying them and interpreting as `ascii`.

### CODE REVIEW

**Review by mollyIV — APPROVED** [review]
Thanks for creating this pull request <3 

I've been testing the changes on one of personal projects and can confirm it's working as expected - I can get the activity log parsed correctly. 

Prior to these changes, I've been getting an error complaining about incorrect activity log file. See https://github.com/MobileNativeFoundation/XCLogParser/issues/219 for more details. 

Can we please make the CI checks green? 🙏 

Other than that, let's ship it :shipit:

**Review by aleksandergrzyb — DISMISSED** [review]

**Comment by frugoman** [comment]
Any reason why this one's not getting attention? does seem like the fix needed.

**Comment by memoto** [comment]
@polac24 @ecamacho @CognitiveDisson Waiting for your approve 🙏

**Comment by aleksandergrzyb** [comment]
There are some linting issues 👀

File: Sources/XCLogParser/loglocation/LogLoader.swift [hunk 1]
```diff
@@ -26,7 +26,17 @@ public struct LogLoader {
         do {
             let data = try Data(contentsOf: url)
             let unzipped = try data.gunzipped()
-            guard let contents =  String(data: unzipped, encoding: .ascii) else {
+            let string: String? = unzipped.withUnsafeBytes { pointer in
+                guard let charPointer = pointer
+                    .assumingMemoryBound(to: CChar.self)
+                    .baseAddress
```
Comment (daeho-ro) [inline Sources/XCLogParser/loglocation/LogLoader.swift:32]:
```suggestion
                    .baseAddress
```

**Comment by memoto** [comment]
@aleksandergrzyb Thanks for suggestion that has fixed the linting!

**Comment by memoto** [comment]
@aleksandergrzyb Your stale review were dismissed due to lint fix, could you please approve again?

**Review by aleksandergrzyb — APPROVED** [review]

**Pull request merged**